packages/vat-data/src/kind-utils.js

```diff
@@ -1,22 +1,25 @@
 import { provide } from '@agoric/store';
-import {
-  defineDurableKind,
-  defineDurableKindMulti,
-  makeKindHandle,
-} from './vat-data-bindings.js';
+import { defineDurableKind, makeKindHandle } from './vat-data-bindings.js';
+
+const { fromEntries, entries } = Object;
+
+export const dropContext =
+  fn =>
+  (_, ...args) =>
+    fn(...args);
+// @ts-expect-error TODO statically recognize harden
+harden(dropContext);
 
 export const provideKindHandle = (baggage, kindName) =>
   provide(baggage, `${kindName}_kindHandle`, () => makeKindHandle(kindName));
 // @ts-expect-error TODO statically recognize harden
 harden(provideKindHandle);
 
-export const provideDurableSingleton = (
-  baggage,
-  kindName,
-  behavior,
-  options = undefined,
-) => {
+export const ProvideFar = (baggage, kindName, methods, options = undefined) => {
```

Comment:
```suggestion
/**
 * @template T
 * @param {import('@agoric/store').MapStore<string, unknown>} baggage
 * @param {string} kindName
 * @param {T} methods
 * @param {import('./types.js').DefineKindOptions<unknown>} [options]
 * @returns {import('@endo/far').ERef<T>}
 */
export const ProvideFar = (baggage, kindName, methods, options) => {
```

Reply:
I did something similar but different at https://github.com/Agoric/agoric-sdk/pull/5674/commits/98a96e947077a098aab3f1f7fcda0d04438edd97

I left the optional parameter syntax on the options argument. I think we should in general.

I replaced the `ERef<T>` with simply `T`. However, I see at https://github.com/endojs/endo/blob/8077e2134615d6a1cbe8c8f5350f664b56f0a17b/packages/marshal/src/make-far.js#L88 the similar `T & RemotableBrand<{}, T>`. Please let me know if I should use this instead.

Attn @michaelfig who authored `RemotableBrand`

PTAL

Reply:
I'm not sure. I need to better understand the remote call and marshalling systems. I do know the types are a loose approximation so far.